apps/devhub/tests/test_views.py

```diff
@@ -2531,6 +2577,52 @@ def test_force_beta(self):
         f = File.objects.all().order_by('-created')[0]
         assert f.status == amo.STATUS_BETA
 
+    @mock.patch('devhub.views.sign_file')
+    def test_unlisted_addon_sideload(self, mock_sign_file):
+        """Sideloadable addons need manual full review."""
+        assert self.addon.status == amo.STATUS_PUBLIC  # Fully reviewed.
+        self.addon.update(is_listed=False, trusted=False)
+        self.post()
+        file_ = File.objects.all().order_by("-created")[0]
+        # The status stays unchanged: it needs a manual full review.
+        assert self.addon.status == amo.STATUS_PUBLIC
+        assert file_.status == amo.STATUS_UNREVIEWED
+        assert not mock_sign_file.called
+
+    @mock.patch('devhub.views.sign_file')
+    def test_unlisted_addon_fail_validation(self, mock_sign_file):
+        """Files that fail validation need a manual preliminary review."""
+        self.addon.update(
+            is_listed=False, status=amo.STATUS_LITE, trusted=False)
+        assert self.addon.status == amo.STATUS_LITE  # Preliminary reviewed.
+        # Make sure the file has validation warnings or errors.
+        self.upload.update(
+            validation='{"notices": 2, "errors": 0, "messages": [],'
+                       ' "metadata": {}, "warnings": 1}')
+        self.post()
+        file_ = File.objects.all().order_by("-created")[0]
+        # The status stays unchanged: it needs a manual preliminary review.
+        assert self.addon.status == amo.STATUS_LITE
+        assert file_.status == amo.STATUS_UNREVIEWED
+        assert not mock_sign_file.called
+
+    @mock.patch('devhub.views.sign_file')
+    def test_unlisted_addon_pass_validation(self, mock_sign_file):
+        """Files that pass validation are automatically signed/reviewed."""
+        self.addon.update(
+            is_listed=False, status=amo.STATUS_LITE, trusted=False)
+        # Make sure the file has no validation warnings nor errors.
+        self.upload.update(
+            validation='{"notices": 2, "errors": 0, "messages": [],'
+                       ' "metadata": {}, "warnings": 0}')
+        assert self.addon.status == amo.STATUS_LITE  # Preliminary reviewed.
+        self.post()
+        file_ = File.objects.all().order_by("-created")[0]
+        # The status is changed to preliminary reviewed and the file is signed.
+        assert self.addon.status == amo.STATUS_LITE
+        assert file_.status == amo.STATUS_LITE
+        assert mock_sign_file.called
```

Comment:
It has no warning, and so the `mock_sign_file` is called.

We'll need to have this rewritten in some way at least when we implement [bug 1153231](https://bugzilla.mozilla.org/show_bug.cgi?id=1153231) (which is the meat of the automatic reviewing/signing).

In this PR I'm only doing it naively to be able to test the full flow: if it passes the validation with NO warning or error, it's automatically reviewed/signed, otherwise it will go through the prelim review queue. This will change, and we'll have a specific (new?) set of checks that need to pass, and a subset of the current warnings/errors that won't prevent the automatic signing.

If it's a sideload addon, it will go through the full review queue (this is the expected behavior).